### PR TITLE
AUTO-187, AUTO-182: Mocked therapy settings

### PR DESCRIPTION
**Pull request by jh-bate**
A big update for the new therapy settings work key points are 

- now uses "mocked therapy settings" as the start point 
- guardrails test language up dated to be consistent 
- guardrails tests broken out to test each part separately. Setting of the value, presence (or absence) of  the warning icon and message
- rework test setup so all values are options passed in. Can now also start in closed loop mode with cgm data 
- general cleanup

### CODE REVIEW

File: src/config/limits.js [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//update based on therapy settings
 var mgdL = {
```
Comment (darinkrauss) [inline src/config/limits.js:2]: FYI - There are tickets coming through that will make the therapy setting warnings and limits different based upon the age (which is calculated from entered birthdate). Something to think about before it happense.

File: src/settings/suspendThresholdScreen.js [hunk 2]
```diff
@@ -30,9 +30,9 @@ class SuspendThresholdScreen extends base.EntryScreen {
     Header() {
         return match.accessible.TextLabel(this.screenText.Header).atIndex(0);
     }
-    async OpenPicker() {
-        await match.accessible.TextLabel(this.bgUnitsLabel).atIndex(0).tap();
-    }
+    // async OpenPicker() {
```
Comment (darinkrauss) [inline src/settings/suspendThresholdScreen.js]: [nit] Commented out? Can it be removed?

File: tests/guardrails/basalRateSchedule.tests.js [hunk 3]
```diff
@@ -1,83 +1,59 @@
+const name = require('./testNames');
+
 module.exports = (test) => {
     var screen;
-    var settings;
+    var therapyScreen;
```
Comment (darinkrauss) [inline tests/guardrails/basalRateSchedule.tests.js]: [nit] `therapySettingsScreen`?

File: tests/guardrails/correctionRangeSchedule.tests.js [hunk 4]
```diff
@@ -1,122 +1,97 @@
+const name = require('./testNames');
+
 module.exports = (test) => {
     var screen;
-    var settings;
+    var therapyScreen;
```
Comment (darinkrauss) [inline tests/guardrails/correctionRangeSchedule.tests.js]: [nit] `therapySettingsScreen`? (Last message about this. Assume this applies to all instances throughout code.)

File: tests/guardrails/suspendThreshold.tests.js [hunk 5]
```diff
@@ -1,90 +1,74 @@
...
+                current: { value: screenLimit.min.limit },
             });
         });
-        it('check for no guardrail warning icon', async () => {
-            await expect(screen.GuardrailWarningIconPicker()).toBeNotVisible();
+        it(name.HasGuardrailIcon, async () => {
+            await expect(screen.GuardrailWarningIconPicker()).toBeVisible();
+        });
+        it(name.HasGuardrailMessage, async () => {
+            await expect(screen.GuardrailMessage('Low Suspend Threshold')).toBeVisible();
```
Comment (darinkrauss) [inline tests/guardrails/suspendThreshold.tests.js]: [nit] Is this really the message "Low Suspend Threshold" or is this a key that maps to the actual message. If the latter, then consider wrapping like we do all other strings. So, something like `screen.LowSuspendThresholdGuardrailMessage()`?

File: tests/guardrails/testNames.js [hunk 6]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
```
Comment (darinkrauss) [inline tests/guardrails/testNames.js]: What is this used for?

**Review by darinkrauss — APPROVED** [review]
LGTM! There are a couple things you might want to address, but I see no issues. 🎃

File: tests/guardrails/testNames.js (hunk 6 above)
Comment (jh-bate) [reply tests/guardrails/testNames.js]: just to keep the naming consistent across all of the guardrails tests ... I would change the naming in one and then it be incorrect in the others

File: src/config/limits.js (hunk 1 above)
Comment (jh-bate) [reply src/config/limits.js:2]: yeah - I am mulling over this in general and was going to have a chat to you. I need dynamic setup based on the therapy settings

**Pull request merged**